core/presentation/designsystem/src/main/java/com/nolawiworkineh/designsystem/components/TaskyToolBar.kt

```diff
@@ -74,109 +73,104 @@ fun TaskyTopAppBar(
+                        )
                     }
+                } else {
+                    navigationIcon?.invoke()
                 }
-            },
+            }
+        },
 
         actions = {
-                Row(
-                    modifier = Modifier
-                        .fillMaxHeight(), // Ensure actions take the full height
-                    verticalAlignment = Alignment.CenterVertically
-                ) {
-                    // If actionIcons are provided, invoke them (e.g., Save, Edit)
-                    actionIcons?.invoke(this)
-
-                    // Handle the dynamic end icon and menu items
-                    if (showEndIcon) {
-                        if (menuItems.isNotEmpty()) {  // Show dropdown only if there are menu items
-                            Box {
-                                // Dropdown menu items
-                                DropdownMenu(
-                                    expanded = isDropDownOpen,
-                                    onDismissRequest = { isDropDownOpen = false }
-                                ) {
-                                    menuItems.forEachIndexed { index, item ->
-                                        Row(
-                                            verticalAlignment = Alignment.CenterVertically,
-                                            modifier = Modifier
-                                                .clickable {
-                                                    onMenuItemClick(index)
-                                                    isDropDownOpen = false
-                                                }
-                                                .fillMaxWidth()
-                                                .padding(horizontal = 16.dp)
-                                        ) {
-                                            item.icon?.let {
-                                                Icon(
-                                                    imageVector = it,
-                                                    contentDescription = item.title,
-                                                    tint = item.tint
-
-                                                )
-                                            }
-                                            Spacer(modifier = Modifier.width(8.dp))
-                                            Text(text = item.title)
+            Row(
+                modifier = Modifier
+                    .fillMaxHeight(), // Ensure actions take the full height
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                // If actionIcons are provided, invoke them (e.g., Save, Edit)
+                actionIcons?.invoke(this)
+
+                // Handle the dynamic end icon and menu items
+                if (endIcon != null) {
+                    endIcon { isDropDownOpen = !isDropDownOpen }
```

Comment:
Because this lambda does not look like a composable function already, having the trailing lambda not contain composable code can lead to some confusion.

Don’t you think the following is more readable and clear?
```
endIcon(
  onClick = { isDropDownOpen = !isDropDownOpen }
)
```